lib/errorHandler.js

```diff
@@ -79,7 +78,16 @@ module.exports = function (config) {
                     body.password = '<omitted>';
                 }
             }
-            logger.error(req.connection.remoteAddress, req.method, req.originalUrl, statusCode, err.stack, err.data, util.inspect(body), util.inspect(req.headers));
+
+            var _logger;
+
+            if (reporter) {
+                _logger = reporter(err, req, res);
+            } else {
+                _logger = logger;
+            }
```

Comment:
Determining which _logger_ is used is done in the [factory](https://github.com/One-com/express-errorhandler/blob/ed6a10a1f6344148153f75f4ba07df9c40c65f36/lib/errorHandler.js#L45-L47).

The reporter is used to overwrite any logger, but needs context from within the function, so I can't assign it from the outside.

An alternative would be:

```js
if (config.reporter) {
  logger = (err, req, res) => config.reporter(err, req, res);
} else if (config.logger) {
  logger = () => config.logger;
} else {
  const _logger = { error: console.log.bind(console) }

  logger = () => _logger;
}

...

return function errorHandler(err, req, res, next) {
  ...

  logger(err, req, res)(req.connection.remoteAddress, req.method, req.originalUrl, statusCode, err.stack, err.data, util.inspect(body), util.inspect(req.headers));

  ...
}
```

Which would be an additional function call for the regular case. I decided for the condition inside the function instead because I find it more readable, but of course both would work. Both are trade-offs.